File: Iniciante/1045 - Tipos de Triângulos/Python/1045.py
def main():
    original_numbers = map(float, input().split())
    ordered_numbers = sorted(original_numbers, reverse=True)
    triangle_type = check_triangle_type(ordered_numbers)
    print(triangle_type.strip())


def check_triangle_type(ordered_numbers):
    formatted_text = ""
    a = ordered_numbers[0]
    b = ordered_numbers[1]
    c = ordered_numbers[2]

    if a >= b + c:
        return "NAO FORMA TRIANGULO\n"

    if a**2 == b**2 + c**2:
        formatted_text += "TRIANGULO RETANGULO\n"
    elif a**2 > b**2 + c**2:
        formatted_text += "TRIANGULO OBTUSANGULO\n"
    elif a**2 < b**2 + c**2:
        formatted_text += "TRIANGULO ACUTANGULO\n"

    if a == b and a == c:
        formatted_text += "TRIANGULO EQUILATERO\n"
    elif a == b or a == c or b == c:
        formatted_text += "TRIANGULO ISOSCELES\n"

    return formatted_text


if __name__ == "__main__":
    main()

File: Iniciante/1045 - Tipos de Triângulos/JavaScript/1045.js
function main() {
  const input =  require("fs").readFileSync("/dev/stdin", "utf8");
  const originalNumbers = input.trim().split(/\s+/).map(Number);
  const orderedNumbers = sortNumbers(originalNumbers);
  const triangleType = checkTriangleType(orderedNumbers);
  console.log(triangleType.trim());
}

function sortNumbers(originalNumbers) {
  const copy = [...originalNumbers];
  copy.sort((a, b) => b - a);
  return copy;
}

function checkTriangleType(orderedNumbers) {
  let formattedText = "";
  const a = orderedNumbers[0];
  const b = orderedNumbers[1];
  const c = orderedNumbers[2];

  if (a >= b + c) {
    return "NAO FORMA TRIANGULO\n";
  }

  if (Math.pow(a, 2) == Math.pow(b, 2) + Math.pow(c, 2)) {
    formattedText += "TRIANGULO RETANGULO\n";
  } else if (Math.pow(a, 2) > Math.pow(b, 2) + Math.pow(c, 2)) {
    formattedText += "TRIANGULO OBTUSANGULO\n";
  } else if (Math.pow(a, 2) < Math.pow(b, 2) + Math.pow(c, 2)) {
    formattedText += "TRIANGULO ACUTANGULO\n";
  }

  if (a == b && a == c) {
    formattedText += "TRIANGULO EQUILATERO\n";
  } else if (a == b || a == c || b == c) {
    formattedText += "TRIANGULO ISOSCELES\n";
  }

  return formattedText;
}

main();
